com.microsoft.mrtk.windowsspeech/Subsystems/TextToSpeech/WindowsTextToSpeechSubsystem.cs

```diff
@@ -0,0 +1,175 @@
+{
+    [Preserve]
+    [MRTKSubsystem(
+        Name = "com.microsoft.mrtk.windowsspeech.texttospeech",
+        DisplayName = "Windows Text-To-Speech Subsystem",
+        Author = "Microsoft",
+        ProviderType = typeof(WindowsTextToSpeechSubsystemProvider),
+        SubsystemTypeOverride = typeof(WindowsTextToSpeechSubsystem),
+        ConfigType = typeof(BaseSubsystemConfig))]
+    public class WindowsTextToSpeechSubsystem : TextToSpeechSubsystem
+    {
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.SubsystemRegistration)]
+        static void Register()
+        {
+            // Fetch subsystem metadata from the attribute.
+            var cinfo = XRSubsystemHelpers.ConstructCinfo<WindowsTextToSpeechSubsystem, TextToSpeechSubsystemCinfo>();
+
+            if (!WindowsTextToSpeechSubsystem.Register(cinfo))
+            {
+                Debug.LogError($"Failed to register the {cinfo.Name} subsystem.");
+            }
+        }
+
+        [Preserve]
+        class WindowsTextToSpeechSubsystemProvider : Provider
+        {
+            public WindowsTextToSpeechSubsystemProvider() : base()
+            { }
+
+            public override void Destroy()
+            {
+#if WINDOWS_UWP
+                if (synthesizer != null)
+                {
+                    synthesizer.Dispose();
+                    synthesizer = null;
+                }
+#endif // WINDOWS_UWP
+            }
+
+            #region ITextToSpeechSubsystem implementation
+
+#if !(WINDOWS_UWP || UNITY_STANDALONE_WIN || UNITY_EDITOR_WIN)
+            private bool haveLogged = false;
+#endif
+
+            /// <inheritdoc/>
+            public override async Task<bool> TrySpeak(string phrase, AudioSource audioSource)
+            {
+                if (audioSource == null)
+                {
+                    Debug.LogError("Must specify the AudioSource object on which the text to speech data should be applied.");
+                    return false;
+                }
+
+                byte[] waveData = await Synthesize(phrase);
+                if (waveData == null)
+                {
+                    return false;
+                }
+
+                // Convert from bytes to floats and create an AudioClip.
+                if (!TextToSpeechHelpers.TryConvertWaveData(
+                        waveData,
+                        out int samples,
+                        out int sampleRate,
+                        out int channels,
+                        out float[] audioFloats))
+                {
+                    Debug.LogError("Failed to convert speech audio format.");
+                    return false;
+                }
+
+                audioSource.clip = TextToSpeechHelpers.CreateAudioClip(
+                    "SynthesizedText",
+                    audioFloats,
+                    samples,
+                    channels,
+                    sampleRate);
+
+                audioSource.Play();
```

Comment:
Do we care about `Play()` vs `PlayOneShot()`? Do we have any guidance on how often (if at all) devs should reuse the audio source they pass into this method?

Reply:
i anticipate that the audio source will be reused by the upcoming self-voiced screen reader in many scenarios, especially when rendering as flat stereo (there would be only one audio source used)